test/unit/streaming.controllers.PlaybackControllers.js

```diff
@@ -328,6 +319,151 @@ describe('PlaybackController', function () {
                 videoModelMock.fireEvent('waiting');
             });
         });
+    });
+
+    describe('start time', function () {
+        let expectedSeekTime;
+        let doneFn;
+
+        let staticStreamInfo = { manifestInfo: { isDynamic: false }, start: 10, duration: 600 };
+        let dynamicStreamInfo = { manifestInfo: { isDynamic: true }, start: 50, duration: Infinity };
+        let dvrWindowRange = { start: 70, end: 100 };
+
+        let onPlaybackSeeking = function (e) {
+            eventBus.off(Events.PLAYBACK_SEEKING, onPlaybackSeeking);
+            expect(e.seekTime).to.equal(expectedSeekTime);
+            doneFn();
+        };
+
+        beforeEach(function () {
+            videoModelMock.time = -1;
+            eventBus.on(Events.PLAYBACK_SEEKING, onPlaybackSeeking, this);
+        });
+
+        it('should start static stream at period start', function (done) {
+            doneFn = done;
+
+            expectedSeekTime = staticStreamInfo.start;
+
+            playbackController.initialize(staticStreamInfo);
+            eventBus.trigger(Events.STREAM_INITIALIZED, {});
+        });
+
+        it('should start static stream at #t', function (done) {
+            doneFn = done;
+
+            let uriStartTime = 10;
+            uriFragmentModelMock.setURIFragmentData({t: uriStartTime.toString()});
+
+            expectedSeekTime = staticStreamInfo.start + uriStartTime;
+
+            playbackController.initialize(staticStreamInfo);
+            eventBus.trigger(Events.STREAM_INITIALIZED, {});
+        });
+
+        it('should start static stream at period start if #t is beyond period end', function (done) {
+            doneFn = done;
+
+            let uriStartTime = 700;
```

Comment:
Suggest to define this as staticStreamInfo.duration + 100 in case we ever change the value in staticStreamInfo.duration